src/frontend/js/types/web-analytics/google_analytics.d.ts

```diff
@@ -0,0 +1,15 @@
+export {};
+
+/**
+ * Declare the Google Analytics ga function.
+ */
+declare global {
+  function ga(
+    tracker: string,
+    hitType: string,
+    eventCategory: string,
+    eventAction: string,
+    eventLabel: string,
+    eventValue?: any,
+  ): void;
+}
```

Comment:
```suggestion
import { Maybe } from 'types/utils'

/**
 * Declare the Google Analytics ga function.
 */
declare global {
  const ga: Maybe<(
    tracker: string,
    hitType: string,
    eventCategory: string,
    eventAction: string,
    eventLabel: string,
    eventValue?: any,
  ): void>;
}
```

Reply:
Can't make it work...

I've changed to:
```ts
/**
 * Declare the Google Analytics ga function.
 */
declare function ga(
  tracker: string,
  hitType: string,
  eventCategory: string,
  eventAction: string,
  eventLabel: string,
  eventValue?: any,
): void;
```

What do you think?

Reply:
I included your suggestion, but it fail when I run frontend testing. Error:
```
js/utils/api/web-analytics/google_analytics.spec.ts:17:12 - error TS2339: Property 'ga' does not exist on type 'typeof globalThis'.

17     global.ga = function (

```

So I changed the `const` to `let` and on the spec changed `global.ga = function (` to `ga = function (`. Now it is ok.

Reply:
Hmmm... Did you try the following in your test
```ts
(global as any).ga = jest.fn();
```

In this way you could keep `ga` as const in our declaration file.

Reply:
Nop. With your suggestion, now it is fine.